### PR TITLE
Certificates import: Add support of pem keychain with private key

### PR DESCRIPTION
**Pull request by proton72**
Private key sometimes required for authentication in web resources across enterprise networks.
 
Currently import of the pem file content support certificates only, ignoring any private key, associated with user certificate.
The reason of that behavior is lack of private keys support in _certutil_. Consequently, trustargs **TCu,Cu,Tu** for the certificate are useless (**TC,C,T** only supported).

Provided solution resolves the issue with steps:

1. export **ca and client certs** from pem file to nss db with _certutil_ 
2. convert **client cert with associated private key** from pem format to p12 format
3. export **private key** to  nss db with _pk12util_

Such 3-steps export provides ability to migrate all certs and keys from pem file to nss db.

### CODE REVIEW

File: static/chrome/entrypoint.sh [hunk 1]
```diff
@@ -41,8 +41,14 @@ if env | grep -q ROOT_CA_; then
   for e in $(env | grep ROOT_CA_ | sed -e 's/=.*$//'); do
     certname=$(echo -n $e | sed -e 's/ROOT_CA_//')
     echo ${!e} | base64 -d >/tmp/cert.pem
-    certutil -A -n ${certname} -t "TCu,Cu,Tu" -i /tmp/cert.pem -d sql:$HOME/.pki/nssdb
+    certutil -A -n ${certname} -t "TC,C,T" -i /tmp/cert.pem -d sql:$HOME/.pki/nssdb
+    if cat tmp/cert.pem | grep -q "PRIVATE KEY"; then
+      AUXILIARY_PASS=123456
```
Comment (vania-pooh) [inline static/chrome/entrypoint.sh:46]: Need to be able to read key password (`AUXILIARY_PASS=123456`) from external environment variable.

File: static/chrome/entrypoint.sh [hunk 2]
```diff
@@ -41,8 +41,14 @@ if env | grep -q ROOT_CA_; then
...
+      openssl pkcs12 -export -in /tmp/cert.pem -clcerts -nodes -out /tmp/key.p12 -passout pass:${AUXILIARY_PASS}
+      pk12util -d sql:$HOME/.pki/nssdb -i /tmp/key.p12 -W ${AUXILIARY_PASS}
+    fi
     rm /tmp/cert.pem
+    rm /tmp/key.p12
```
Comment (vania-pooh) [inline static/chrome/entrypoint.sh:51]: Shouldn't it be inside `if`?

**Review by vania-pooh — CHANGES_REQUESTED** [review]
Also need an example command of how to generate such certificates, so we could test.

File: static/chrome/entrypoint.sh (hunk 2 above)
Comment (proton72) [reply static/chrome/entrypoint.sh]: It looks reasonable. Updated in b507b78

**Comment by vania-pooh** [comment]
@proton72 still need to do something with `AUXILIARY_PASS` variable.

File: static/chrome/entrypoint.sh (hunk 1 above)
Comment (proton72) [reply static/chrome/entrypoint.sh]: The actual purpose of AUXILIARY_PASS is providing consistent fake password for _openssl -export_ and _pk12util -i_ commands - nothing more. The variable AUXILIARY_PASS is replaced with EMPTY_PASS in 54e2bc8, which could be a little more clear.
Btw, some _*.pem_ files could have password-protected private keys, and such files are not supported in current PR.

**Comment by proton72** [comment]
> Also need an example command of how to generate such certificates, so we could test.

Command for generation self-signed certificate with private key is:
`openssl req -x509 -newkey rsa:4096 -out cert-and-key.pem -keyout cert-and-key.pem -nodes`
You will be prompted to enter some cert info, like country code and owner email.

**Comment by vania-pooh** [comment]
@proton72 ok, but would you change with EMPTY_PASSWORD work in case when certificate private key is password protected?

**Comment by proton72** [comment]
> @proton72 ok, but would you change with EMPTY_PASSWORD work in case when certificate private key is password protected?

New env variable `PRIVATE_KEY_PASS` supported in 4ffbd62 with default value "empty string", which works properly even in case of _pem_ file without password protection.

Please use snippet below to generate _pem_ file with `lorem` password for private key:
`openssl req -x509 -newkey rsa:4096 -out cert-and-key-with-pass.pem -passout pass:lorem -keyout cert-and-key-with-pass.pem`

**Review by vania-pooh — APPROVED** [review]

**Comment by vania-pooh** [comment]
@proton72 thank you for contribution!

**Pull request merged**